### ISSUE TITLE
Javadocs don't report invalid at clauses for specific types

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2812]
TestClass.java
````
package test;

/**
 * TestClass.
 * 
 * @return This doesn't exist.
 * @throws Exception
 *             This doesn't exist.
 */
public class TestClass {
    /**
     * field.
     * 
     * @return This doesn't exist.
     * @throws Exception
     *             This doesn't exist.
     */
    public String field;
}
````
config.xml
````
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />

	<module name="TreeWalker">
		<module name="JavadocType">
			<property name="scope" value="public" />
		</module>
		<module name="JavadocVariable">
			<property name="scope" value="public" />
		</module>
	</module>
</module>
````
output from: `java -jar checkstyle-6.14.1-all.jar -c config.xml TestClass.java`
````
Starting audit...
Audit done.
````
Expected output:
````
Starting audit...
[ERROR] TestClass.java:6:4: Invalid 'return' tag for 'TestClass'.
[ERROR] TestClass.java:7:4: Invalid 'throws' tag for 'TestClass'.
[ERROR] TestClass.java:14:8: Invalid 'return' tag for 'field'.
[ERROR] TestClass.java:15:8: Invalid 'throws' tag for 'field'.
Audit done.
Checkstyle ends with 1 errors.
````

None of the current Javadoc Checks report issues were invalid tags are used for specific types. Classes and fields don't throw exceptions or return data, only methods do.

**Comment by rnveach** [comment]
@romani Do you think this should be a new module and not added onto existing checks?

**Comment by romani** [comment]
I think that such validation should be done by existing Checks.

**Comment by aayushRedHat** [comment]
@romani Can i take this issue

**Comment by AayushSaini101** [comment]
The solution of this issue will be using most of the implementation of the other JavadocMethodCheck, but most of the methods are private, Is it good to implement again the implementation in the other class ? Will it cause any kind of quality issue @romani @rnveach @nrmancuso

**Comment by nrmancuso** [comment]
We need to reconsider our position on a “fix” for this issue, last activity in here was six years ago.

**Comment by AayushSaini101** [comment]
> We need to reconsider our position on a “fix” for this issue, last activity in here was six years ago.

Yes this is an issue I checked

**Comment by nrmancuso** [comment]
> > We need to reconsider our position on a “fix” for this issue, last activity in here was six years ago.
> 
> Yes this is an issue I checked

Please restate, I do not understand this statement. To be clear, I am asking that we revisit the proposed solution to extend existing checks, and come up with a clear plan before starting work on this issue. If we want to extend two checks to provide this functionality, we should have a separate issue for each.

**Comment by AayushSaini101** [comment]
> > > We need to reconsider our position on a “fix” for this issue, last activity in here was six years ago.
> > 
> > 
> > Yes this is an issue I checked
> 
> Please restate, I do not understand this statement. To be clear, I am asking that we revisit the proposed solution to extend existing checks, and come up with a clear plan before starting work on this issue. If we want to extend two checks to provide this functionality, we should have a separate issue for each.

@nrmancuso  yes we can extend the existing checks for this issue but I have one doubt The solution of this issue will be using most of the implementation of the other JavadocMethodCheck, but most of the methods are private, Is it good to implement again the implementation in the other class ? Will it cause any kind of quality issue

**Comment by romani** [comment]
I am ok to start new Check for this, what is name we can use for it? 
Check will be ast based.
Can someone provide design? What properties will be used?

**Comment by AayushSaini101** [comment]
> I am ok to start new Check for this, what is name we can use for it? Check will be ast based. Can someone provide design? What properties will be used?

@romani Why we need new check for this ? we have existing check JavadocType, JavadocVariable . We can improve them

**Comment by romani** [comment]
Existing Checks are not ast based, so implementation will never be stable, and will be very error prone, so spending time in what is not going to be good is not reasonable.

Additionally they are mega checks that covering too much and and when users need customization it becomes a big problem of design as properties becomes to conflict, and design becomes problematic and user are frustrated.

Better to do Check that do some limited scope but do this extremely well.

We need something like UnusedJvadocTag UnrelevantJavadocTag.

**Comment by AayushSaini101** [comment]
> Existing Checks that you mentioned are not ast based, so implementation will never be stable, and will be very error prone, so spending time in what is not going to be good is not reasonable.
> 
> Additionally they are mega checks that covering too much and and when users need customization it becomes a big problem of design as properties becomes to conflict, and design becomes problematic and user are frustrated.
> 
> Better to do Check that do some limited scope but do this extremely well.
> 
> We need something like UnusedJvadocTag UnrelevantJavadocTag.

I am ready to create new check for the unused tags. Can I create new issue for this ?

**Comment by romani** [comment]
No need for new issue, we already have this issue.
Please read https://checkstyle.org/report_issue.html#How_to_request_new_Check.2FModule.3F and look at referenced example issue as example of what is expected to share to minimize conversations

**Comment by romani** [comment]
Now all Checks are migrated to AST.
Existing Checks already very complicated.
Let's create new Check , InappropriateJavadocBlockTagsOnXxxxx, on method, on type, on field, on package.

We need to create 4 issues.

**Comment by romani** [comment]
3 issues are created, no issue for  InappropriateJavadocBlockTagsOnMethod as I am not sure that tag is not applicable to methods, so no need to make it for now, users will open issue ones something is detected.
